tests/conftest.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+from flask import Flask
+from flask_meld import Meld
+
+
+@pytest.fixture(scope='module')
+def app(tmpdir_factory):
+    # create directory structure of project/meld/components
+    app_dir = tmpdir_factory.mktemp('meld')
+    app_dir.mkdir('components')
+    module = app_dir.join("search.py")
+    with module.open('w') as f:
+        class_def = ["from flask_meld.component import Component",
+                     "class Search(Component):",
+                     "\tstate=''"]
+        f.writelines(f"{line}\n" for line in class_def)
+
+    meld = Meld()
+    app = Flask(__name__)
+    app.config["MELD_COMPONENT_DIR"] = app_dir
+    app.secret_key = __name__
+    meld.init_app(app)
+    return app
+
+
+@pytest.fixture(scope='module')
+def app_factory(tmpdir_factory):
```

Comment:
seems a lot of this is duplicated, can you dedup / abstract it into another fixture for reuse?

Reply:
btw what is the difference conceptually here between `app` and `app_factory`?

Reply:
There are generally two ways of creating a flask application. You can create a global Flask object or you can create the Flask object within a function. The factory method is more commonly used for anything but the most simple application. However, I wanted to support both. More concretely for this test case:
Factory application is loaded within a function in `project/app/__init__.py` vs just having an `app.py` in the root project directory.

Reply:
Further reference if you are interested: https://flask.palletsprojects.com/en/1.1.x/tutorial/factory/

Reply:
Awesome link and approach, I will use this moving forward. Thanks!